azurerm/internal/services/servicebus/resource_arm_servicebus_namespace_network_rule.go

```diff
@@ -0,0 +1,308 @@
+package servicebus
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/servicebus/mgmt/2018-01-01-preview/servicebus"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/set"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	validateNetwork "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/servicebus/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/servicebus/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceServiceBusNamespaceNetworkRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServiceBusNamespaceNetworkRuleCreateUpdate,
+		Read:   resourceServiceBusNamespaceNetworkRuleRead,
+		Update: resourceServiceBusNamespaceNetworkRuleCreateUpdate,
+		Delete: resourceServiceBusNamespaceNetworkRuleDelete,
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.ServiceBusNamespaceNetworkRuleID(id)
+			return err
+		}),
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"namespace_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ServiceBusNamespaceName,
+			},
+
+			"default_action": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(servicebus.Allow),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(servicebus.Allow),
+					string(servicebus.Deny),
+				}, false),
+			},
+
+			"ip_masks": {
```

Comment:
I’d expect this variable to be `ip_rules`, based on other `network_rules`. Any reason to prefer `ip_masks`?

Reply:
Well, this is the REST API property name. And I am wondering does `ip_masks` usually mean these are the IPs that is allowed or in the opposite?
There is also an API for individually modifying this `CreateOrUpdateIPFilterRules`, but I did not take this name, since this name implies that those IPs are filtered out, but actually based on what is showing on the portal, these IPs should be the IPs that are allowed to pass.

Reply:
`ip_masks` confused me at first glance, so I would argue for `ip_rules`. But what can I say, I opened a PR for a solved problem 🙄

Reply:
`ip_rules` works for me! I will change this right away